File: leetcode/easy/70.Climbing Stairs/python/climbingStairs.py
"""Climbing Stairs"""

# Approach 4: Fibonacci Number
def climbing_stairs(n):
    if n == 1:
        return 1
    first = 1
    second = 2
    for i in range(3, n + 1):
        third = first + second
        first = second
        second = third
    return second

# Testing
n0 = 2
n1 = 3
print(climbing_stairs(7))

'''Complexity Analysis:
Time Complexity: O(N)
Space Complexity: O(1)'''

File: leetcode/easy/70.Climbing Stairs/js/climbingStairs.js
"use strict";

// Approach 4: Fibonacci Number

/**
 * 
 * @param {number} n
 * @return {number} 
 */
var climbingStairs = function(n) {
    if (n === 1) {
        return 1;
    };
    let first = 1;
    let second = 2;
    for (let i = 3; i < n + 1; i++) {
        let third = first + second;
        first = second;
        second = third;
    };
    return second;
};

// Complexity Analysis:
// Time Complexity: O(N)
// Space Complexity: O(1)